source/extensions/common/wasm/wasm_vm.h

```diff
@@ -0,0 +1,199 @@
+using WasmCall3Void = std::function<void(Context*, Word, Word, Word)>;
+using WasmCall4Void = std::function<void(Context*, Word, Word, Word, Word)>;
+using WasmCall5Void = std::function<void(Context*, Word, Word, Word, Word, Word)>;
+using WasmCall6Void = std::function<void(Context*, Word, Word, Word, Word, Word, Word)>;
+using WasmCall7Void = std::function<void(Context*, Word, Word, Word, Word, Word, Word, Word)>;
+using WasmCall8Void = std::function<void(Context*, Word, Word, Word, Word, Word, Word, Word, Word)>;
+using WasmCall0Word = std::function<Word(Context*)>;
+using WasmCall1Word = std::function<Word(Context*, Word)>;
+using WasmCall2Word = std::function<Word(Context*, Word, Word)>;
+using WasmCall3Word = std::function<Word(Context*, Word, Word, Word)>;
+using WasmCall4Word = std::function<Word(Context*, Word, Word, Word, Word)>;
+using WasmCall5Word = std::function<Word(Context*, Word, Word, Word, Word, Word)>;
+using WasmCall6Word = std::function<Word(Context*, Word, Word, Word, Word, Word, Word)>;
+using WasmCall7Word = std::function<Word(Context*, Word, Word, Word, Word, Word, Word, Word)>;
+using WasmCall8Word = std::function<Word(Context*, Word, Word, Word, Word, Word, Word, Word, Word)>;
+#define FOR_ALL_WASM_VM_EXPORTS(_f)                                                                \
+  _f(WasmCall0Void) _f(WasmCall1Void) _f(WasmCall2Void) _f(WasmCall3Void) _f(WasmCall4Void)        \
+      _f(WasmCall5Void) _f(WasmCall8Void) _f(WasmCall0Word) _f(WasmCall1Word) _f(WasmCall3Word)
+
+// Calls out of the WASM VM.
+// 1st arg is always a pointer to raw_context (void*).
+using WasmCallback0Void = void (*)(void*);
+using WasmCallback1Void = void (*)(void*, Word);
+using WasmCallback2Void = void (*)(void*, Word, Word);
+using WasmCallback3Void = void (*)(void*, Word, Word, Word);
+using WasmCallback4Void = void (*)(void*, Word, Word, Word, Word);
+using WasmCallback5Void = void (*)(void*, Word, Word, Word, Word, Word);
+using WasmCallback6Void = void (*)(void*, Word, Word, Word, Word, Word, Word);
+using WasmCallback7Void = void (*)(void*, Word, Word, Word, Word, Word, Word, Word);
+using WasmCallback8Void = void (*)(void*, Word, Word, Word, Word, Word, Word, Word, Word);
+using WasmCallback0Word = Word (*)(void*);
+using WasmCallback1Word = Word (*)(void*, Word);
+using WasmCallback2Word = Word (*)(void*, Word, Word);
+using WasmCallback3Word = Word (*)(void*, Word, Word, Word);
+using WasmCallback4Word = Word (*)(void*, Word, Word, Word, Word);
+using WasmCallback5Word = Word (*)(void*, Word, Word, Word, Word, Word);
+using WasmCallback6Word = Word (*)(void*, Word, Word, Word, Word, Word, Word);
+using WasmCallback7Word = Word (*)(void*, Word, Word, Word, Word, Word, Word, Word, Word);
+using WasmCallback8Word = Word (*)(void*, Word, Word, Word, Word, Word, Word, Word, Word, Word);
+using WasmCallback9Word = Word (*)(void*, Word, Word, Word, Word, Word, Word, Word, Word, Word,
```

Comment:
Why only this have 9 param version?

Reply:
We use the 9 Word version for a call out of the VM but for calls into the VM we use at most 8 Words.